### PR TITLE
Feign parameter #value() fallback

### PR DESCRIPTION
**Pull request by mbenson**
Builds on #828 to provide parameter name fallback on Java 8 with `-parameters` compile flag (bytecode-based parameter name discovery not applicable for `abstract` methods). This may bear some discussion because the implementation provided assumes that the parameter name is an acceptable fallback for any `String value()` annotation element that is defaulted.

### CODE REVIEW

File: spring-cloud-netflix-core/pom.xml [hunk 1]
```diff
@@ -199,4 +199,25 @@
 			<optional>true</optional>
 		</dependency>
 	</dependencies>
+	<profiles>
+		<profile>
+			<id>java8plus</id>
+				<activation>
+					<jdk>[1.8,2.0)</jdk>
```
Comment (spencergibb) [inline spring-cloud-netflix-core/pom.xml]: We build with java8 by default, but have to have java 7 compatibility. I'm guessing this breaks that?

**Comment by mbenson** [comment]
I went to some lengths not to break Java 7 compatibility, hence the reflection-based strategy for enabling the related unit test. That Maven profile is enabled when running on JDK >= 1.8 and in that case compiles with the `-parameters` argument, but this doesn't seem to create problematic bytecode. My assumption is that the Java 7 bytecode structure already included whatever mechanism is used in Java 8 to hold the parameter names.

File: spring-cloud-netflix-core/pom.xml (hunk 1 above)
Comment (mbenson) [inline spring-cloud-netflix-core/pom.xml]: I went to some lengths not to break Java 7 compatibility, hence the reflection-based strategy for enabling the related unit test. That Maven profile is enabled when running on JDK >= 1.8 and in that case compiles with the -parameters argument, but this doesn't seem to create problematic bytecode. My assumption is that the Java 7 bytecode structure already included whatever mechanism is used in Java 8 to hold the parameter names.

Comment (spencergibb) [inline spring-cloud-netflix-core/pom.xml]: OK, we require a java 8 build anyway (one optional module requires it), so this will always be true.

**Pull request closed without merging**